packages/flame/lib/src/game/mixins/single_game_instance.dart

```diff
@@ -0,0 +1,16 @@
+import '../../components/component.dart';
+import 'game.dart';
+
+mixin SingleGameInstance on Game {
```

Comment:
This will differentiate cases with a single and multiple `GameWidget`s in the Flutter tree, this should not be the case.
I think we'll just have to make the `staticGameInstance` non-static, and then traverse the tree each time that a component is added.

Reply:
This is an _optional_ mixin: if the user knows that the application that they are building only ever uses a single `Game` instance, within a single `GameWidget`, they can opt into using this mixin.